DirectProgramming/DPC++FPGA/ReferenceDesigns/cholesky_inversion/README.md

```diff
@@ -0,0 +1,279 @@
+Similarly, the streaming_cholesky_inversion.hpp linear algebra header library implements the inversion of matrices with pipe interfaces, based on the outputs of the cholesky decomposition implemented in streaming_cholesky.hpp.
+Both the decomposition and the inversion have template parameters that can be set to decompose/invert custom sized real or complex square matrices.
+This DPC++ reference design demonstrates the use of these Cholesky decomposition and inversion header libraries on 32 * 32 real matrices.
+
+
+***Documentation***:  The [DPC++ FPGA Code Samples Guide](https://software.intel.com/content/www/us/en/develop/articles/explore-dpcpp-through-intel-fpga-code-samples.html) helps you to navigate the samples and build your knowledge of DPC++ for FPGA. <br>
+The [oneAPI DPC++ FPGA Optimization Guide](https://software.intel.com/content/www/us/en/develop/documentation/oneapi-fpga-optimization-guide) is the reference manual for targeting FPGAs through DPC++. <br>
+The [oneAPI Programming Guide](https://software.intel.com/en-us/oneapi-programming-guide) is a general resource for target-independent DPC++ programming.
+
+| Optimized for                     | Description
+---                                 |---
+| OS                                | Linux* Ubuntu* 18.04/20.04, RHEL*/CentOS* 8, SUSE* 15; Windows* 10
+| Hardware                          | Intel® Programmable Acceleration Card (PAC) with Intel Arria® 10 GX FPGA <br> Intel® FPGA Programmable Acceleration Card (PAC) D5005 (with Intel Stratix® 10 SX)
+| Software                          | Intel® oneAPI DPC++ Compiler <br> Intel® FPGA Add-On for oneAPI Base Toolkit
+| What you will learn               | Implementing high performance Cholesky matrix decomposition on FPGAs.
+| Time to complete                  | 1 hr (not including compile time)
+
+
+**Performance**
+Please refer to the performance disclaimer at the end of this README.
+
+| Device                                            | Throughput
+|:---                                               |:---
+| Intel® PAC with Intel Arria® 10 GX FPGA           | 215k matrices/s for real matrices of size 32 * 32
+| Intel® FPGA PAC D5005 (with Intel Stratix® 10 SX) | 255k matrices/s for real matrices of size 32 * 32
+
+
+## Purpose
+
+This FPGA reference design demonstrates Cholesky-based matrix inversion, a common operation employed in linear algebra, through use of the streaming_cholesky.hpp and the streaming_cholesky_inversion.hpp header libraries.
+The Cholesky decomposition takes a hermitian, positive-definite matrix _A_ (input) and computes the lower triangular matrix _L_ such that: _L_ * _L*_ = A, where _L*_ in the conjugate transpose of _L_.
+Given this decomposition, the inverse of A can be computed as A<sup>-1</sup> = (_L_ * (_L*_))<sup>-1</sup> = (_L*_)<sup>-1</sup> * _L_<sup>-1</sup> = ((_L_<sup>-1</sup>)* ) * _L_<sup>-1</sup>.
```

Comment:
I would put this math in a separate block (apparently github markdown now supports [LaTEX](https://github.blog/2022-05-19-math-support-in-markdown/))
```suggestion
Given this decomposition, the inverse of A can be computed as A<sup>-1</sup> = (_L_ * (_L*_))<sup>-1</sup> = (_L*_)<sup>-1</sup> * _L_<sup>-1</sup> = ((_L_<sup>-1</sup>)* ) * _L_<sup>-1</sup>.
```
```suggestion
Given this decomposition, the inverse of A can be computed as 

$$A^{-1} = (L * L^{*})^{-1}$$

$$= L^{*-1} * L^{-1}$$

$$= ((L^{-1*} ) * L^{-1}$$
```

$$A^{-1} = (L * L^{*})^{-1}$$

$$= L^{*-1} * L^{-1}$$

$$= ((L^{-1*} ) * L^{-1}$$


Also, the math seems a bit off to me; I would write as:

$$A^{-1} = (L * L^{*})^{-1}$$

$$= L^{-1} * L^{*-1}$$

$$= L^{-1} * (L^{-1})^{*}$$

Reply:
I did not know that github had support for latex syntax, that's cool! 
I think the math is ok, as 
$$(AB)^{-1} = B^{-1}A^{-1}